src/components/VillagerDetails.js

```diff
@@ -0,0 +1,98 @@
+import React, { useEffect } from 'react';
+import { useParams, Link } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faArrowLeft } from '@fortawesome/free-solid-svg-icons';
+import { getOneVillager } from '../redux/villagers/villagers';
+
+const VillagerDetails = () => {
+  const villagerArr = [];
+  const { id } = useParams();
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (villagerArr.length === 0) {
+      dispatch(getOneVillager(id));
+    }
+  }, []);
+
+  const villager = useSelector((state) => state.villagersReducer);
+  if (villager.name === undefined) {
+    return (
+      <div className="container p-4">
+        <div className="row">
+          <div className="col-sm-4" />
+          <div className="col-sm-8">
+            <h1 className="text-white">
+              Loading
+            </h1>
+          </div>
+        </div>
+      </div>
+    );
+  }
+  return (
+    <div className="text-white">
+      <div className="row headbanner pt-1">
+        <Link
+          className="nav-link"
+          to="/villagers"
+        >
+          <div className="row pl-2 text-white">
+            <h2 className="links-nav col">
+              <FontAwesomeIcon icon={faArrowLeft} />
+            </h2>
+            <h2 className="col">Villager Details</h2>
+          </div>
+        </Link>
+      </div>
+      <div className="row firstCont">
+        <div className="col-xs-6 p-4 villagerCont">
+          { /* eslint-disable dot-notation */
+            <img src={villager['image_uri']} alt="villager" className="villagerPic" />
```

Comment:
- [ ] Kindly enable the eslint here. The following line of code works just fine anyway.
```
image={villager.image_uri}
```